rest-client-common/src/main/java/io/apicurio/rest/client/auth/OidcAuth.java

```diff
@@ -70,7 +73,14 @@ private void requestAccessToken() {
             ).collect(Collectors.joining("&"));
             final AccessTokenResponse accessTokenResponse = apicurioHttpClient.sendRequest(TokenRequestsProvider.obtainAccessToken(paramsEncoded));
             this.cachedAccessToken = accessTokenResponse.getToken();
-            this.cachedAccessTokenExp = System.currentTimeMillis() / 1000 + accessTokenResponse.getExpiresIn();
+            /**
+             * expiresIn is in seconds
+             */
+            long expiresIn = accessTokenResponse.getExpiresIn();
+            if (expiresIn > TOKEN_EXPIRATION_REDUCTION_SEC) {
```

Comment:
This is not going to work. `getExpiresIn` is going to give you the lifespan value for the access token. For mas, that will be 300 so you're not gonna enter the if in any case. And, actually, if you enter the if, you're gonna expire the token in that exact moment since you're going to set the expiration to the past.

Reply:
ok , `TOKEN_EXPIRATION_REDUCTION_SEC` equals to 2000 was a mistake , but apart from that I think the logic here makes sense